### ISSUE TITLE
Model Produced by R-Package has Changed

### ISSUE THREAD

**Issue by darkmobility** [microsoft/LightGBM#2168]
I have a LightGBM code which was running well on the Kaggle Santander Customer Prediction data sets on April 22, however when I tried to run it again today, my outputs were drastically different. 
Previously, I could make my trees normally but now I keep getting the error "stopped training because there are no more leaves that meet the split requirements" and my trees keep stopping at 1 or 2 leaves.

 Can anyone please help me on this? I am suspecting that there are some changes in the R package but I am not sure why. The data files can be found here https://www.kaggle.com/c/santander-customer-transaction-prediction.

[Kaggle Project (2).zip](https://github.com/microsoft/LightGBM/files/3165864/Kaggle.Project.2.zip)

Here is my code for your reference and thank you very much!

**Comment by jameslamb** [comment]
Thanks for reporting and for using LightGBM, @darkmobility .

Could you please:

1. Share the output of running `sessionInfo()` in R (in a session where your code is running)
2. Tell us how you installed the `lightgbm` R package (precompiled, built from source with `build_r.R`, etc.)
3. Share the exact code you are running (and the full output log if you have it)
4. Change the title of this issue to something like "model produced by R-package has changed" or something else that differentiates this issue from others in our backlog

Those items will help us to understand what version you are running and allow us to check if we can reproduce the issue you're seeing.

Thanks!

**Comment by darkmobility** [comment]
Thank you for your reply.

1) Here is my sessioninfo.
![image](https://user-images.githubusercontent.com/43195500/57568838-4dd31c80-741f-11e9-810c-6260c01e7b42.png)
2) I downloaded the lightgbm via conda install on this website link. https://anaconda.org/conda-forge/lightgbm
3) Here is my code for your reference.
library(e1071)
library(caTools)
library(Matrix)
setwd("C:/Users/Benjamin/Desktop/")

library(lightgbm)

train = read.csv("train.csv", header = TRUE)
test = read.csv("test.csv", header = TRUE)

lgb.train= lgb.Dataset(data=Matrix(data=as.matrix(train[,-1]), sparse=TRUE), 
                       label=Matrix(data=as.matrix(train$target), sparse=TRUE))

bst = lightgbm(data = lgb.train,num_leaves = 10, learning_rate = 0.1, nrounds = 840, objective = "binary", 
               verbose=2,boosting_type="goss",min_data_in_leaf=1000,max_depth=8, min_gain_to_split=2.0,
              tree_learner="data")

pred = predict(bst, Matrix(as.matrix(test), sparse = TRUE))

write.csv(pred, "results_new.csv")

Thank you very much and I look forward to your reply. :)

**Comment by guolinke** [comment]
@darkmobility do you use the same built of LightGBM to run your code?

>  keep getting the error "stopped training because there are no more leaves that meet the split requirements" and my trees keep stopping at 1 or 2 leaves.

From this description, I think it is more like a data issue.

**Comment by StrikerRUS** [comment]
Unfortunately, we do not maintain LightGBM on conda, so it's hard for us to tell you what changes were included at a  particular date. If it's still an issue for you, I think it's better to ask package maintainers at conda.
